api/v1alpha1/openapi.yaml

```diff
@@ -4086,6 +4087,16 @@ components:
               description: Reference to the container image for the application package.
           required:
             - image
+    ArtifactApplicationProviderSpec:
+      type: object
+      allOf:
+        - type: object
+          properties:
+            artifact:
```

Comment:
@fzdarsky I was thinking it would be better to be explicit here instead of trying to discover image vs artifact. Its going to make the API a lot saner to implement. Auto discovery adds a lot of complexity. But curious your thoughts.

Reply:
I was assuming auto discover boils down to fetching the referenced manifest, checking whether that contains the `artifactType` field. If so, it's an artifact of the specified type (e.g. `application/vnd.cncf.helm.chart.v2`). If not, fetch the config object and check whether its `mediaType` is `application/vnd.oci.image.config.v1+json` (= container image) or not (= artifact).

Would fields other than the `artifact` or `image` field change or would the _service_ (who may not be able to fetch the manifest) do things differently depending on artifact or image? If not, I'd lean towards keeping things simple for the API user.